src/gnc/orbit_controller.cpp

```diff
@@ -25,56 +25,188 @@
+  auto &that_h_ecef0 = state.that_h_ecef0;
+  auto &this_h_hat = state.this_h_hat;
+
+  auto &DCM_hill_ecef0 = state.DCM_hill_ecef0;
+
+  // Move to the inertial ECEF0 frame
+  {
+    lin::Vector3f w_earth;
+    env::earth_angular_rate(data.t, w_earth);
+
+    // Position in ECEF and ECEF0 are the same
+    this_r_ecef0 = data.r_ecef;
+    that_r_ecef0 = data.r_ecef + data.dr_ecef;
+    this_r_hat = this_r_ecef0 / lin::norm(this_r_ecef0);
+
+    // Velocities need to velocity of the ECEF frame itself added on
+    this_v_ecef0 = data.v_ecef - lin::cross(w_earth, this_r_ecef0);
+    that_v_ecef0 = data.v_ecef + data.dv_ecef - lin::cross(w_earth, that_r_ecef0);
+    this_v_hat = this_v_ecef0 / lin::norm(this_v_ecef0);
+
+    // Calculate the satellites angular momentums (without scaling by mass)
+    this_h_ecef0 = lin::cross(this_r_ecef0, this_v_ecef0);
+    that_h_ecef0 = lin::cross(that_r_ecef0, that_v_ecef0);
+    this_h_hat = this_h_ecef0 / lin::norm(this_h_ecef0);
+  }
+
+  // Calculate the DCM rotating from ECEF0 to the hill frame
+  utl::dcm(DCM_hill_ecef0, that_r_ecef0, that_v_ecef0);
+
+  // In-plane controller
+  lin::Vector3d dv_in_plane;
+  {
+    // Calculate orbital energies
+    double const this_energy = energy(this_r_ecef0, this_v_ecef0);
+    double const that_energy = energy(that_r_ecef0, that_v_ecef0);
+
+    // Angular rate of that satellite's hill frame
+    lin::Vector3d const w_hill = that_h_ecef0 / lin::fro(that_r_ecef0);
+
+    // Position and velocity of this satellite in the other's hill frame
+    lin::Vector3d const r_hill = DCM_hill_ecef0 * (this_r_ecef0 - that_r_ecef0).eval();
+    lin::Vector3d const v_hill = DCM_hill_ecef0 * (this_v_ecef0 - that_v_ecef0).eval() - lin::cross(w_hill, r_hill);
+
+    // Hill frame PD controller
+    dv_in_plane = this_v_hat * (
+        ( K_p * r_hill(1)) +                  // Hill position term
+        (-K_d * v_hill(1)) +                  // Hill velocity term
+        (-K_e * (this_energy - that_energy))  // Energy term
+      );
+  }
+
+  // Angular momentum controller (matching orbital planes)
+  lin::Vector3d dv_plane;
+  {
+    // Project that satellite's angular momentum onto the plane perpendicular
+    // to this satellite's position vector
+    lin::Vector3d that_h_proj = that_h_ecef0 - lin::dot(that_h_ecef0, this_r_hat) * this_r_hat;
+
+    // Calculate the angle between this projection of the other satellites
+    // angular momentum and our angular momentum
+    double const theta = lin::atan2(
+        lin::dot(that_h_proj, lin::cross(this_r_ecef0, this_h_ecef0)),
+        lin::dot(that_h_proj, this_h_ecef0)
+      ); // ^^ this seems a tad weird to me and would be of very different
+         // orders of magnitude - i.e. `this_r_ecef0` or `this_r_hat`?
+
+    dv_plane = this_h_hat * (K_h * theta / mass);
+    // Don't understand why we divide by mass ^^ as we never were working with
+    // orbital angular momentum anyway
+  }
```

Comment:
yeah I just copied whatever Stewart did in his test file lol. I think @kkrol27 you and I were talking about this, but I could not really follow what he was doing either haha. but based on the graphs generated, whatever he's doing seems to work, so I didn't question it too much. but if any of you have a better way to do this, I'm all ears

Reply:
I'm not too worried about it either as it may just be a technicality that changes our gain for this term. Just left it in there in case people had the same question in the future.